src/Spice86.Core/Emulator/InterruptHandlers/Input/Mouse/MouseDriver.cs

```diff
@@ -63,16 +66,13 @@ public class MouseDriver : IMouseDriver {
     /// <inheritdoc />
     public int CurrentMinY { get; set; }
 
-    /// <inheritdoc />
-    public IAsmUserRoutineHandler? UserRoutineHandler { set; private get; }
-
     /// <inheritdoc />
     public void BeforeUserHandlerExecution() {
```

Comment:
```suggestion
    public void HandlePS2MouseDataUpdate() {
```
The purpose of the call to this method is to let the mouse driver know that the PS/2 mouse device has new data.
We currently only have 1 mouse device (PS/2) and that is cheating by not putting the data on the I/O ports but exposing it through public properties.
If we want more accurate emulation and more device types we are going to need to handle input from more sources, triggered by other things than irq12/int74.
Only after processing this data, and making it available to int33 queries, the driver should check if someone else has expressed an interest in mouse events by registering a usercallback. Only then, if the registration matches the current event, should a call be made to the user code.